compilers/concrete-optimizer/concrete-optimizer/src/dag/unparametrized.rs

```diff
@@ -243,6 +243,36 @@ impl OperationDag {
         self.add_lut(rounded, table, out_precision)
     }
 
+    pub(crate) fn get_input_index_iter(&self) -> impl Iterator<Item = usize> + '_ {
+        self.operators
+            .iter()
+            .enumerate()
+            .filter_map(|(index, op)| match op {
+                Operator::Input { .. } => Some(index),
+                _ => None,
+            })
+    }
+
+    pub(crate) fn get_output_index(&self) -> Vec<usize> {
```

Comment:
There is already a code doing that in analyse.
https://github.com/zama-ai/concrete/blob/main/compilers/concrete-optimizer/concrete-optimizer/src/optimization/dag/solo_key/analyze.rs#L209.